lib/services/collections_service.dart

```diff
@@ -315,11 +324,38 @@ class CollectionsService {
   }
 
   String decryptCollectionPath(Collection collection) {
+    final key = collection.attributes.version != null
+        ? getCollectionKey(collection.id)
+        : _config.getKey();
     return utf8.decode(CryptoUtil.decryptSync(
         Sodium.base642bin(collection.attributes.encryptedPath),
-        _config.getKey(),
+        key,
         Sodium.base642bin(collection.attributes.pathDecryptionNonce)));
   }
+
+  Collection getcollectionWithDecryptedName(Collection collection) {
```

Comment:
Instead of this, just have a 
```dart
String _getDecryptedCollectionName(Collection collection)
```
That way a new `Collection` object does not have to be instantiated.

Reply:
will have to remove the final attribute from the collection name then